Route/CmsPageRouter.php

```diff
@@ -207,7 +210,25 @@ protected function generateFromPage(PageInterface $page, array $parameters = arr
             throw new \RuntimeException(sprintf('Page "%d" has no url or customUrl.', $page->getId()));
         }
 
-        return $this->decorateUrl($url, $parameters, $referenceType);
+        if (!$this->context instanceof SiteRequestContext) {
+            @trigger_error(
+                'The context should implement SiteRequestContextInterface on the next release',
```

Comment:
Sorry, but this message is not precise enough. Bear in mind that Symfony will show it without any stack trace.
This would be better:

```
Since, 3.x, when calling '.__METHOD__.', '. __CLASS__ . '::$context should implement SiteRequestContextInterface. '.
'This will become mandatory in 4.x'
```